src/tree/azure/AzureResourceTreeDataProviderBase.ts

```diff
@@ -50,20 +48,32 @@ export abstract class AzureResourceTreeDataProviderBase extends ResourceTreeData
         return super.isAncestorOf(element, id)
     }
 
-    protected async getAzureAccountExtensionApi(): Promise<AzureSubscriptionProvider> {
+    protected async getAzureSubscriptionProvider(): Promise<AzureSubscriptionProvider> {
         // override for testing
         if (ext.testing.overrideAzureSubscriptionProvider) {
             return ext.testing.overrideAzureSubscriptionProvider();
         } else {
             if (!this.subscriptionProvider) {
                 this.subscriptionProvider = await ext.subscriptionProviderFactory();
-                await this.subscriptionProvider.waitForFilters();
             }
 
-            this.filtersSubscription = this.subscriptionProvider.onFiltersChanged(() => this.notifyTreeDataChanged());
-            this.statusSubscription = this.subscriptionProvider.onStatusChanged(() => this.notifyTreeDataChanged());
+            this.statusSubscription = vscode.authentication.onDidChangeSessions((evt: vscode.AuthenticationSessionsChangeEvent) => {
```

Comment:
We could add an event to the shared package to mark sign in. I don't feel strongly about it being here.